flexget/plugins/sites/freshon.py

```diff
@@ -0,0 +1,188 @@
+from __future__ import unicode_literals, division, absolute_import
+from builtins import *  # noqa pylint: disable=unused-import, redefined-builtin
+
+import logging
+import re
+import urllib
+
+from flexget import plugin
+from flexget.entry import Entry
+from flexget.event import event
+from flexget.utils.soup import get_soup
+from flexget.utils.search import normalize_unicode
+
+log = logging.getLogger('search_freshon')
+
+
+class SearchFreshon(object):
+
+    schema = {
+        'type': 'object',
+        'properties':
+        {
+            'username': {'type': 'string'},
+            'password': {'type': 'string'},
+            'passkey': {'type': 'string'},
+            'category': {
+                'type': 'string',
+                'default': 'all',
+                'enum': ['hd', 'webdl', 'all']
+            },
+            'freeleech': {
+                'type': 'string',
+                'default': 'all',
+                'enum': ['free', 'half', 'all']
```

Comment:
Best to use this as a dict for reusability:
```python
LEECHSTATUS = {
    'all': 0,
    'free': 3,
    'half': 4
}
```
then in the schema use:
```python
'enum': list(LEECHSTATUS.keys())
```
and later you can access this dict too